tools/ld-analyse/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+  HINTS ${QT_LIBRARY_DIR}
+)
+
+set ( QWT_LIBRARIES ${QWT_LIBRARY} )
+
+include ( FindPackageHandleStandardArgs )
+find_package_handle_standard_args( Qwt REQUIRED_VARS QWT_LIBRARY QWT_INCLUDE_DIR VERSION_VAR QWT_VERSION_STRING )
+
+if (Qwt_FOUND AND NOT TARGET Qwt::Qwt)
+  add_library(Qwt::Qwt UNKNOWN IMPORTED)
+  set_target_properties(Qwt::Qwt PROPERTIES
+                        INTERFACE_INCLUDE_DIRECTORIES "${QWT_INCLUDE_DIRS}"
+                        IMPORTED_LOCATION "${QWT_LIBRARIES}")
+endif ()
+
+mark_as_advanced (
+  QWT_LIBRARY
+  QWT_INCLUDE_DIR
+)
+
+#
+#
+#
+#
+#
+#
+#
+#
+# Things above this line can go to an external file if possible
+# Anything beneath stays here
+#
+#
+#
+#
+#
+#
+#
+#
+
+cmake_minimum_required(VERSION 3.22)
+project(ld-analyse)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+#list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_modules")
+
+# Set up AUTOMOC and some sensible defaults for runtime executionGui
+# When using Qt 6.3, you can replace the code block below with
+#qt_standard_project_setup()
+set(CMAKE_AUTOMOC ON)
+include(GNUInstallDirs)
+set(CMAKE_AUTOUIC ON)
+
+find_library(QWT_LIBRARY_RELEASE NAMES qwt qwt-qt5 qt5-qwt PATH_SUFFIXES lib)
+find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core Gui Widgets HINTS ${QT_LIBRARY_DIR})
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Gui Widgets)
+
+# Name all the files
+add_executable(ld-analyse MACOSX_BUNDLE
+    blacksnranalysisdialog.cpp blacksnranalysisdialog.h blacksnranalysisdialog.ui
+    busydialog.cpp busydialog.h busydialog.ui
+    closedcaptionsdialog.cpp closedcaptionsdialog.h closedcaptionsdialog.ui
+    main.cpp
+    mainwindow.cpp mainwindow.h mainwindow.ui
+    oscilloscopedialog.cpp oscilloscopedialog.h oscilloscopedialog.ui
+    vectorscopedialog.cpp vectorscopedialog.h vectorscopedialog.ui
+    aboutdialog.cpp aboutdialog.h aboutdialog.ui
+    videoparametersdialog.cpp videoparametersdialog.h videoparametersdialog.ui
+    chromadecoderconfigdialog.cpp chromadecoderconfigdialog.h chromadecoderconfigdialog.ui
+    tbcsource.cpp tbcsource.h
+    vbidialog.cpp vbidialog.h vbidialog.ui
+    configuration.cpp configuration.h
+    dropoutanalysisdialog.cpp dropoutanalysisdialog.h dropoutanalysisdialog.ui
+    ../ld-chroma-decoder/palcolour.cpp ../ld-chroma-decoder/palcolour.h
+    ../ld-chroma-decoder/comb.cpp ../ld-chroma-decoder/comb.h
+    ../ld-chroma-decoder/componentframe.cpp ../ld-chroma-decoder/componentframe.h
+    ../ld-chroma-decoder/outputwriter.cpp ../ld-chroma-decoder/outputwriter.h
+    ../ld-chroma-decoder/transformpal.cpp ../ld-chroma-decoder/transformpal.h
+    ../ld-chroma-decoder/transformpal2d.cpp ../ld-chroma-decoder/transformpal2d.h
+    ../ld-chroma-decoder/transformpal3d.cpp ../ld-chroma-decoder/transformpal3d.h
+    ../ld-chroma-decoder/framecanvas.cpp ../ld-chroma-decoder/framecanvas.h
+    ../ld-chroma-decoder/sourcefield.cpp ../ld-chroma-decoder/sourcefield.h
+    ../library/filter/firfilter.h
```

Comment:
The stuff you've built as libraries shouldn't be listed here - you will get these by linking with the library.

Reply:
- **Libraries and filenames:** Compiles and runs
- **Libraries only:** Undefined references to functions when attempting to compile
- **Filenames only:** Cannot compile because it can't find certain files

TL;DR both libraries and filenames are required to compile